### PR TITLE
Add contribution docs

### LINKED ISSUE #264: lists: Paragraphs that are part of lists should be full width
Anything that's meant to be "responsive" when inside a list is getting it's width reduced twice (paragraphs, codeblocks etc.)
Screen Shot 2019-07-18 at 3 13 13 PM
Screen Shot 2019-07-18 at 3 07 08 PM

### PR DESCRIPTION
**Pull request by vpicone**
closes #264 

Adds documentation on getting started developing with the theme.

### CODE REVIEW

File: packages/example/src/pages/contributing.mdx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+   ```sh
+   git remote -v
+   ```
+
+## Development
+
+We use [yarn](https://yarnpkg.com/lang/en/docs/install/) and yarn workspaces to develop the Carbon Gatsby theme. This allows us to have a development environment that's closely linked to the theme with minimal setup. Run `yarn install` to install all of the projects dependencies.
+
+This project has two packages: the actual theme package (`gatsby-theme-carbon`) and and the `example` package. The example package emulates a project which uses the theme. It's only dependencies are Gatsbt, React and the adjacent theme package. The `example` package also serves as the theme's documentation and [website](https://gatsby-theme-carbon.now.sh); it's deployed on every merge to master.
```
Comment (abbeyhrt) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
This project has two packages: the actual theme package (`gatsby-theme-carbon`) and and the `example` package. The example package emulates a project which uses the theme. It's only dependencies are Gatsby, React and the adjacent theme package. The `example` package also serves as the theme's documentation and [website](https://gatsby-theme-carbon.now.sh); it's deployed on every merge to master.
```

File: packages/example/src/pages/contributing.mdx [hunk 2]
```diff
@@ -0,0 +1,175 @@
+---
+title: Contributing
+---
+
+<PageDescription>
+
+This guide is meant for developers who would like to contribute to the theme. We'll show how to get the development environment set up as quickly as possible.
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
Use this guide to contribute to the theme. We'll show you how to get the development environment set up as quickly as possible so you can start contributing.
```

File: packages/example/src/pages/contributing.mdx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+</PageDescription>
+
+<AnchorLinks>
+  <AnchorLink>Project Setup</AnchorLink>
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
  <AnchorLink>Project setup</AnchorLink>
```

File: packages/example/src/pages/contributing.mdx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  <AnchorLink>Development</AnchorLink>
+  <AnchorLink>Work in a branch</AnchorLink>
+  <AnchorLink>Sass and CSS Modules</AnchorLink>
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
  <AnchorLink>Sass and CSS modules</AnchorLink>
```

File: packages/example/src/pages/contributing.mdx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  <AnchorLink>VS Code</AnchorLink>
+</AnchorLinks>
+
+## Project Setup
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
## Project setup
```

File: packages/example/src/pages/contributing.mdx [hunk 6]
```diff
@@ -0,0 +1,175 @@
+---
+title: Contributing
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
title: Contributions
```
We are using `contributions` on the Carbon site, we should keep them consistent.

File: packages/example/src/data/nav-items.yaml [hunk 7]
```diff
@@ -4,6 +4,12 @@
 - title: Demo
   pages:
     - path: /demo
+- title: Gallery
+  pages:
+    - path: /gallery
+- title: Contributing
```
Comment (connor-leech) [inline packages/example/src/data/nav-items.yaml]:
```suggestion
- title: Contributions
```

Comment (connor-leech) [reply packages/example/src/data/nav-items.yaml]: You'll probably have to rename the file, too. This is what we're calling it on the Carbon website, should keep it consistent.

File: packages/example/src/pages/contributing.mdx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+   cd gatsby-theme-carbon
+   ```
+
+1. When you clone your forked repo the `origin` is set to your fork by default. You'll want to add a remote that points to the upstream repo.
+
+   ```sh
+   git remote add upstream git@github.com:carbon-design-system/gatsby-theme-carbon.git
+   ```
+
+1. In your terminal, verify the remotes have been set
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
1. In your terminal, verify that the remotes have been set
```

Comment (connor-leech) [reply packages/example/src/pages/contributing.mdx]: I never thought I'd suggest someone add "that" to a sentence, but here we are 😂

File: packages/example/src/pages/contributing.mdx [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+   ```sh
+   git remote -v
+   ```
+
+## Development
+
+We use [yarn](https://yarnpkg.com/lang/en/docs/install/) and yarn workspaces to develop the Carbon Gatsby theme. This allows us to have a development environment that's closely linked to the theme with minimal setup. Run `yarn install` to install all of the projects dependencies.
+
+This project has two packages: the actual theme package (`gatsby-theme-carbon`) and and the `example` package. The example package emulates a project which uses the theme. It's only dependencies are Gatsby, React and the adjacent theme package. The `example` package also serves as the theme's documentation and [website](https://gatsby-theme-carbon.now.sh); it's deployed on every merge to master.
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
This project has two packages: the actual theme package (`gatsby-theme-carbon`) and the `example` package. The example package emulates a project which uses the theme. Its only dependencies are Gatsby, React and the adjacent theme package. The `example` package also serves as the theme's documentation and [website](https://gatsby-theme-carbon.now.sh); it's deployed on every merge to master.
```

File: packages/example/src/pages/contributing.mdx [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+git checkout -b my-branch-name
+```
+
+When you're ready to open a pull request, push to your origin remote.
+
+```sh
+git push origin my-branch-name
+```
+
+You'll get a message in your terminal with a url to open up a pull request in the upstream repository. Navigate to that URL and be sure to give a detailed summary of your pull request in the title and body section of the form.
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
You'll get a message in your terminal with a URL to open up a pull request in the upstream repository. Navigate to that URL and be sure to give a detailed summary of your pull request in the title and body section of the form.
```

File: packages/example/src/pages/contributing.mdx [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+## Sass and CSS Modules
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
## Sass and CSS modules
```

File: packages/example/src/pages/contributing.mdx [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+
+For internal theme components we use [Sass](https://sass-lang.com/documentation/syntax) and [CSS modules](https://github.com/css-modules/css-modules#css-modules). This allows us to take advantage of the carbon design system resources while not worrying about className collisions. By default, every Sass file will have access to all of the carbon-components [vars](https://github.com/carbon-design-system/carbon/blob/master/packages/components/src/globals/scss/_vars.scss): color, spacing, theme, and motion as well as type and layout mixins are included.
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
For internal theme components we use [Sass](https://sass-lang.com/documentation/syntax) and [CSS modules](https://github.com/css-modules/css-modules#css-modules). This allows us to take advantage of the Carbon Design System resources while not worrying about className collisions. By default, every Sass file will have access to all of the Carbon components [vars](https://github.com/carbon-design-system/carbon/blob/master/packages/components/src/globals/scss/_vars.scss): color, spacing, theme, and motion, as well as type and layout mixins, are included.
```

Comment (connor-leech) [reply packages/example/src/pages/contributing.mdx]: @vpicone Take another look at this. How should we reference the Sass vars? carbon-components is deprecated

File: packages/example/src/pages/contributing.mdx [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+git push origin my-branch-name
+```
+
+You'll get a message in your terminal with a url to open up a pull request in the upstream repository. Navigate to that URL and be sure to give a detailed summary of your pull request in the title and body section of the form.
+
+## Sass and CSS modules
+
+For internal theme components we use [Sass](https://sass-lang.com/documentation/syntax) and [CSS modules](https://github.com/css-modules/css-modules#css-modules). This allows us to take advantage of the carbon design system resources while not worrying about className collisions. By default, every Sass file will have access to all of the carbon-components [vars](https://github.com/carbon-design-system/carbon/blob/master/packages/components/src/globals/scss/_vars.scss): color, spacing, theme, and motion as well as type and layout mixins are included.
+
+You should co-locate your stylesheet with the component(s) that import it. If the component is `TreeView` then the stylesheeet should be `TreeView.module.scss`. All contained within the `TreeView` directory.
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
You should colocate your stylesheet with the component(s) that import it. If the component is `TreeView` then the stylesheet should be `TreeView.module.scss`. All contained within the `TreeView` directory.
```

File: packages/example/src/pages/contributing.mdx [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+
+### CSS Modules
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
### CSS modules
```

File: packages/example/src/pages/contributing.mdx [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+
+You don't need to prefix your classes, CSS modules will generate the unique class name for you. Import the class from the `.scss` file.
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
You don't need to prefix your classes as CSS modules will generate the unique class name for you. Import the class from the `.scss` file.
```

File: packages/example/src/pages/contributing.mdx [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+
+```scss
+:global(.bx--grid) .codeBlock {
+  @include carbon--type-style('code-01');
+}
+```
+
+## VS Code
+
+To get linting error feedback while writing javascript and scss in VSCode, install the [stylelint](https://marketplace.visualstudio.com/items?itemName=shinnn.stylelint) and [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) extensions. We use ESLint's prettier rules to format and lint all of our javascript in one pass. To get your code to format properly on save, add the following to a `.vscode/settings.json` in the root of your project
```
Comment (connor-leech) [inline packages/example/src/pages/contributing.mdx]:
```suggestion
To get linting error feedback while writing javascript and SCSS in VS Code, install the [stylelint](https://marketplace.visualstudio.com/items?itemName=shinnn.stylelint) and [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) extensions. We use ESLint's prettier rules to format and lint all of our javascript in one pass. To get your code to format properly on save, add the following to a `.vscode/settings.json` in the root of your project
```

**Comment by connor-leech** [comment]
Be sure to change the file names/YAML file to "Contributions" instead of Contributing to stay in line with the Carbon site.

**Pull request merged**